tests/js/object-tests.js

```diff
@@ -515,5 +515,30 @@ module.exports = {
         TestCase.assertEqual(realm.objects('Date')[1].currentDate.getTime(), -10000);
         TestCase.assertEqual(realm.objects('Date')[2].currentDate.getTime(), 1000000000000);
         TestCase.assertEqual(realm.objects('Date')[3].currentDate.getTime(), -1000000000000);
+    },
+    
+    testInheritance: function() {
+        // the equivalent of `class InheritingObject extends Realm.Object`
+        function InheritingObject() {
```

Comment:
Exactly, so I seriously doubt many people are using this relatively undocumented "feature", so I definitely suggest dropping it and maybe in the future provide an optional special method that could be called whenever a new Realm object is created if somebody really badly needs this functionality.